src/mode.rs

```diff
@@ -42,6 +49,8 @@ pub enum Mode {
     Yolo {
         /// When to run rustfix on the test
         rustfix: RustfixMode,
+        /// Whether to write back missing annotations to the test file.
+        write_back: Option<WriteBackLevel>,
```

Comment:
I wonder if we can make this a standalone emitter instead. Should have all the information available, and can then be combined with an existing emitter, run on its own, or even turned into a runtime switchable mode, all without affecting the core ui_test engine at all

Reply:
I've been trying to understand `ui_test` and these changes. Based on my current (most likely limited) understanding, I think that the emitter would be the wrong place or at least it would be inconsistent with how the normal "bless" works.

Instead, I would modify the [`check_annotations`](https://github.com/oli-obk/ui_test/blob/2d077c71f6665f17a10dc80fa6553fd6bec65940/src/lib.rs#L1272) function and have it add the annotations if the related configuration is enabled. This would be inline with how [`check_output`](https://github.com/oli-obk/ui_test/blob/2d077c71f6665f17a10dc80fa6553fd6bec65940/src/lib.rs#L1440) works.

One problem is the current size of `check_annotations`. It's like 200-300 lines. I will look into this approach a bit more. Any thoughts or comments on this suggestion would be appreciated!